libass/ass.c

```diff
@@ -811,7 +809,7 @@ static int process_events_line(ASS_Track *track, char *str)
 
         return process_event_tail(track, event, str, 0);
     } else {
-        ass_msg(track->library, MSGL_V, "Not understood: '%.30s'", str);
+        ass_msg(track->library, MSGL_V, "Not understood: '%*s'", (int)str.len, str.str);
```

Comment:
Also `%.*s`. And if we want exact equivalent then it should be `(int) FFMIN(str.len, 30)` too.

Reply:
Sure, I guess it's not absurd to limit how much garbage this can spew on corrupt input.